sds_data_manager/stacks/processing_stack.py

```diff
@@ -33,6 +33,10 @@ def __init__(
         instrument_target: str,
         instrument_sources: str,
         repo: ecr.Repository,
+        batch_security_group: ec2.SecurityGroup,
```

Comment:
this constructor has a lot of inputs, does it make sense to combine any of them into a `NamedTuple` or something?